pkg/framework/plugins/defaultevictor/defaultevictor.go

```diff
@@ -228,8 +229,28 @@ func (d *DefaultEvictor) PreEvictionFilter(pod *v1.Pod) bool {
 			klog.InfoS("pod does not fit on any other node because of nodeSelector(s), Taint(s), or nodes marked as unschedulable", "pod", klog.KObj(pod))
 			return false
 		}
+	}
+
+	if d.args.NamespaceLabelSelector == nil {
 		return true
 	}
+
+	// check pod by namespace label filter
+	indexName := "namespaceWithLabelSelector"
+	indexer, err := getNamespacesListByLabelSelector(indexName, d.args.NamespaceLabelSelector, d.handle)
```

Comment:
The indexer needs to be created before any of the plugins start running. Otherwise, the cache might not get populated properly. In `New` function.

Reply:
Thank you, i'll try to fix that.